### PR TITLE
filter out of date permit applications for notification link

### PR DESCRIPTION
**Pull request by bkmaibach**
## Description

Refactor search filters in order to provide a link for template update notifications 

## What type of PR is this? (check all applicable)

- [ x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📦 Chore (Release)
- [ ] ✅ Test

## Related Tickets & Documents

https://hous-bssb.atlassian.net/browse/BPHH-1612

## Steps to QA

Update template as super admin
get notifcation as submitter
click the Go link
You should see only the relevant draft permits with yellow highlighting
reset the filter to see everything again.

### CODE REVIEW

File: app/frontend/lib/create-search-model.ts [hunk 1]
```diff
@@ -9,11 +9,9 @@ interface IFetchOptions {
   countPerPage?: number
 }
 
-const filterableStatuses = Object.values(EPermitApplicationStatus)
-export type TFilterableStatus = (typeof filterableStatuses)[number]
-
 export const createSearchModel = <TSortField, TFetchOptions extends IFetchOptions = IFetchOptions>(
-  fetchDataActionName: string
+  fetchDataActionName: string,
+  setFiltersName: string
```
Comment (taw27) [inline app/frontend/lib/create-search-model.ts:14]: make the second field optional as not all search models need filters.

File: app/frontend/stores/requirement-block-store.ts [hunk 2]
```diff
@@ -123,6 +123,10 @@ export const RequirementBlockStoreModel = types
       }
       return response.ok
     }),
+    setRequirementBlockFilters(queryParams) {
+      // Set filters here once necessary
+      // See permit-application-store.ts for example
+    },
```
Comment (taw27) [inline app/frontend/stores/requirement-block-store.ts:129]: idea the create searchModel should handle this if possible. Just make the filters param optional in the create function and escape filter related logic based on that. As not all searches will have filters

File: app/frontend/stores/requirement-template-store.ts [hunk 3]
```diff
@@ -79,6 +79,10 @@ export const RequirementTemplateStoreModel = types
       }
       return response.ok
     }),
+    setRequirementTemplateFilters(queryParams) {
+      // Set filters here once necessary
+      // See permit-application-store.ts for example
+    },
```
Comment (taw27) [inline app/frontend/stores/requirement-template-store.ts:85]: same comment as before

File: app/frontend/stores/user-store.ts [hunk 4]
```diff
@@ -142,6 +142,10 @@ export const UserStoreModel = types
       }
       return response.ok
     }),
+    setUserFilters(queryParams) {
+      // Set filters here once necessary
+      // See permit-application-store.ts for example
+    },
```
Comment (taw27) [inline app/frontend/stores/user-store.ts:148]: same comment as before

**Review by taw27 — CHANGES_REQUESTED** [review]

File: app/frontend/stores/requirement-block-store.ts (hunk 2 above)
Comment (bkmaibach) [reply app/frontend/stores/requirement-block-store.ts:129]: I'm not sure I understand, it does handle this (in the same way it handles fethData using fetchDataActionName)

Comment (taw27) [reply app/frontend/stores/requirement-block-store.ts:129]: @bkmaibach I am saying this shouldn't be required as not all searches use filtering

**Review by taw27 — APPROVED** [review]

**Pull request merged**